examples/with-aptos/src/index.ts

```
@@ -0,0 +1,142 @@
+  );
+
+  // Prepare the transaction payload
+  const token = new TxnBuilderTypes.TypeTagStruct(
+    TxnBuilderTypes.StructTag.fromString("0x1::aptos_coin::AptosCoin")
+  );
+
+  const entryFunctionPayload =
+    new TxnBuilderTypes.TransactionPayloadEntryFunction(
+      TxnBuilderTypes.EntryFunction.natural(
+        "0x1::coin",
+        "transfer",
+        [token],
+        [
+          BCS.bcsToBytes(
+            TxnBuilderTypes.AccountAddress.fromHex(recipientAddress)
+          ),
+          BCS.bcsSerializeUint64(Number(amount)),
+        ]
+      )
+    );
+
+  // Get account sequence number and chain ID
+  const [{ sequence_number: sequenceNumber }, chainId] = await Promise.all([
+    client.getAccount(aptosAddress),
+    client.getChainId(),
+  ]);
+
+  // Build the raw transaction
+  const rawTxn = new TxnBuilderTypes.RawTransaction(
+    TxnBuilderTypes.AccountAddress.fromHex(aptosAddress),
+    BigInt(sequenceNumber),
+    entryFunctionPayload,
+    2000n, // Max gas amount
+    100n, // Gas price per unit
+    BigInt(Math.floor(Date.now() / 1000) + 600), // Expiration timestamp
+    new TxnBuilderTypes.ChainId(Number(chainId))
+  );
+
+  // Get the signing message
+  const signingMessage = TransactionBuilder.getSigningMessage(rawTxn);
+
+  // Sign the transaction using Turnkey with HASH_FUNCTION_NO_OP
+  const txSignResult = await turnkeyClient.apiClient().signRawPayload({
+    signWith: aptosAddress,
+    payload: bytesToHex(signingMessage),
+    encoding: "PAYLOAD_ENCODING_HEXADECIMAL",
+    hashFunction: "HASH_FUNCTION_NOT_APPLICABLE",
+  });
+
+  // Extract r and s from the result
+  const { r, s } = txSignResult;
+
+  // Ensure r and s are 64 hex characters (32 bytes)
+  const rHex = r.padStart(64, "0");
+  const sHex = s.padStart(64, "0");
+
+  // Concatenate r and s to form the signature
+  const txSignatureHex = rHex + sHex;
+
+  // Validate signature length
+  if (txSignatureHex.length !== 128) {
+    console.error(
```

we could just throw an error here

```
throw new Error("Invalid signature length for Ed25519. Expected 128 hex characters.") 
```

and the handler (`main().catch()...`) will catch the error and exit